src/HeatMap/components/HeatMapContainer.ts

```diff
@@ -39,13 +40,29 @@ export default class HeatMapContainer extends Component<HeatMapContainerProps, H
         this.resetSubscriptions(newProps.mxObject);
         if (!this.state.alertMessage) {
             this.fetchData(newProps.mxObject);
+            if (newProps.mxObject) {
+                this.setRefreshInterval(newProps.refreshInterval);
+            }
         }
     }
 
     componentWillUnmount() {
         if (this.subscriptionHandle) {
             window.mx.data.unsubscribe(this.subscriptionHandle);
         }
+        if (this.intervalID) {
+            clearInterval(this.intervalID);
```

Comment:
Global scoped variables should be named explicit.
`window.clearInterval`

Reply:
Added